### PR TITLE
Added validation example for Problem 1.2 of Class (2009)

### PR DESCRIPTION
**Pull request by francoishamon**
This PR adds a validation case for the second problem (Problem 1.2) proposed in the benchmark paper: "A benchmark study on problems related to CO 2 storage in geologic formations", Class et al, 2009.

I got the following result, which is the best I could do given the differences in the other codes' results and the information provided in the paper.

match_problem_1_2

This PR requires some code fixes in the CO2-brine fluid package that will be done in a separate PR.

This PR will require a rebaseline to add a smoke test based on a coarse version of the problem. Rebaseline done in https://github.com/GEOSX/integratedTests/pull/274.

### CODE REVIEW

File: src/docs/sphinx/advancedExamples/validationStudies/thermalLeakyWell/Example.rst [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+   Due to the nonlinear dependence of brine density on temperature, this block does not exactly impose a Dirichlet pressure equal to the initial condition. Instead, here, we impose a linear pressure gradient along the z-axis, whose minimum and maximum values are the same as in the initial state. We could have imposed Dirichlet boundary conditions preserving the initial condition using as many points in `zlin.geos` as there are cells along the z-axis (instead of just two points).  
+
+The **SourceFlux** is the same as in the previous benchmark case (see :ref:`ExampleIsothermalLeakyWell`).
+
+---------------------------------
+Inspecting results
+---------------------------------
+
+We request VTK-format output files and use Paraview to visualize the results.
+The following figures show the distribution of CO2 saturation and pressure along the slice defined by x = 0 at t = 300 days.
```
Comment (joshua-white) [inline src/docs/sphinx/advancedExamples/validationStudies/thermalLeakyWell/Example.rst:236]: 200 or 300?  figure captions say 200 days.

**Review by joshua-white — APPROVED** [review]
Very nice

**Review by herve-gross — APPROVED** [review]
Very nice indeed. Looking forward to continuing this example with thermal effects included.

File: src/docs/sphinx/advancedExamples/validationStudies/thermalLeakyWell/Example.rst (hunk 1 above)
Comment (francoishamon) [reply src/docs/sphinx/advancedExamples/validationStudies/thermalLeakyWell/Example.rst]: Yes, you are right, it should be 200 days

**Pull request merged**